packages/net/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+mod provider;
+
+use std::ops::Deref;
+use url::Url;
+
+pub use http::Method;
+
+pub use provider::*;
+
+#[cfg(any(feature = "blocking", feature = "non_blocking"))]
+const USER_AGENT: &str = "Mozilla/5.0 (X11; Linux x86_64; rv:60.0) Gecko/20100101 Firefox/81.0";
+
+pub trait NetProvider<T> {
```

Comment:
Can we move the `NetProvider` and `RequestHandler` traits into a new `blitz-traits` crate? That would allow `blitz-dom` to depend on the trait without depending on `blitz-net` (and vice versa). There are a couple of other types and traits that could also move into that crate (for similar reasons), but I think that would be best done as a separate PR.

I believe this would also allow us to avoid feature flags and have sync and async net providers in entirely separate crates.